test-cypress/integration/setup/namespaces.spec.js

```diff
@@ -0,0 +1,359 @@
+        cy.createRepository({id: repositoryId});
+        cy.presetRepositoryCookie(repositoryId);
+        cy.warmRepositoryNamespaces(repositoryId);
+
+        cy.visit('/namespaces');
+
+        waitUntilPageIsLoaded();
+    });
+
+    afterEach(() => {
+        cy.deleteRepository(repositoryId);
+    });
+
+    function waitUntilPageIsLoaded() {
+        // Workbench loading screen should not be visible
+        cy.get('.ot-splash').should('not.be.visible');
+
+        // No active loader
+        cy.get('.ot-loader').should('not.be.visible');
+
+        getAddNamespaceForm().should('be.visible');
+    }
+
+    const DEFAULT_NAMESPACES = {
+        'gn': 'http://www.geonames.org/ontology#',
+        'owl': 'http://www.w3.org/2002/07/owl#',
+        'rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+        'rdfs': 'http://www.w3.org/2000/01/rdf-schema#',
+        'wgs': 'http://www.w3.org/2003/01/geo/wgs84_pos#',
+        'xsd': 'http://www.w3.org/2001/XMLSchema#'
+    };
+
+    it('verify initial state', () => {
+        getNoNamespacesAlert().should('not.be.visible');
+
+        // Should be able to insert new prefix
+        getNamespacePrefixField().should('be.visible').and('not.be.disabled');
+        getNamespaceValueField().should('be.visible').and('not.be.disabled');
+        getAddNamespaceButton().should('be.visible').and('not.be.disabled');
+
+        // Should render a table with some default namespaces
+        getNamespacesTable().should('be.visible');
+        getNamespaces().should('have.length', Object.keys(DEFAULT_NAMESPACES).length);
+
+        // Should provide pagination options
+        getNamespacesPerPageMenu().within(() => {
+            // Should show all namespaces by default (they are only 6 so they can be visualized all at once)
+            cy.get('.dropdown-toggle')
+                .should('contain', 'All')
+                .click();
+            cy.get('.page-size-option')
+                .should('have.length', 1)
+                .and('contain', 'All');
+            // Close the menu to avoid overlapping other elements
+            cy.get('.dropdown-toggle').click();
+        });
+
+        // Should show summary of results
+        getNamespacesHeaderPaginationInfo()
+            .should('be.visible')
+            .and('contain', 'Showing 1 - 6 of 6 results');
+
+        // Both header & footer pagination must be the same
+        getNamespacesHeaderPagination()
+            .should('be.visible')
+            .find('li')
+            // Single page + First & Last buttons
+            .should('have.length', 3);
+        getNamespacesPagination()
+            .should('be.visible')
+            .find('li')
+            // Single page + First & Last buttons
+            .should('have.length', 3);
+
+        // Verify default namespaces are present and interactable
+        const defaultPrefixes = Object.keys(DEFAULT_NAMESPACES);
+        getNamespaces().each(($row, $index) => {
```

Comment:
Not sure if it is such a great idea to test that exactly these namespaces are available after repo creation.

Reply:
If you do not change the ruleset when you create repository, the namespaces will be the same, so I think it is okay to test exactly these.